### PR TITLE
I made some small changes and added a 'Bitarray' for more efficient storage of bloom_filter

### PR DESCRIPTION
**Pull request by Lasseone**

### CODE REVIEW

File: bloomfilter.c [hunk 1]
```diff
@@ -24,28 +26,28 @@ int modular_hash(const char string[], int seed_1, int seed_2)
  * @param bloomFilter: the filter to modify
  * @param string: the string to add to the bloom filter.
  */
-void add_string(bool bloom_filter[M], const char string[])
+void add_string(Bitarray* bloom_filter, const char string[])
```
Comment (JosephSBoyle) [inline bloomfilter.c]: **question (non-blocking):** Should the parameter description mention that the arg is a pointer or is that redundant due to the header containing that information?

File: bitarray.c [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+bool read_bit(const Bitarray* bitarray, size_t indx)
+{
+	size_t long_offset = indx / BITS_PER_LONG;
+	unsigned char bit_offset = indx % BITS_PER_LONG;
+	long mask = 1 << (BITS_PER_LONG - bit_offset - 1);
+
+	return (bitarray->data[long_offset] & mask) != 0;
+}
```
Comment (JosephSBoyle) [inline bitarray.c]: **suggestion (blocking):** Is there a description we can add to explain the operations that are happening in these methods? A link to an explanation would suffice also.

File: bloomfilter.c [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
  * @param a string (char array) to hash
  * @param R: a seed value, should be a prime
  * @param M: a seed value, should also be a prime
  */
-static int modular_hash(const char string[], int seed_1, int seed_2)
+int modular_hash(const char string[], int seed_1, int seed_2)
 {
     int hash = 0;
-    for (char character = *string; character != '\0'; character = *++string)
+    for (const char* character = string; *character != '\0'; character++)
```
Comment (JosephSBoyle) [inline bloomfilter.c:18]: **question (non-blocking):** Some questions for my understanding:

1. `const char string[]` an array of pointers to characters, and in line 18 we iterate through each character by incrementing the pointer until the value of `character` (the dereferenced pointer) is equal to the terminating '\0' ? 

2. Is `const char* character = string` initializing `character` as the first element of `string`?

3. `character++` Is this how we get the next char pointer?

File: main.c [hunk 4]
```diff
@@ -17,15 +19,15 @@ int main(int argc, char *argv[])
...
-        add_string(bloom_filter, argv[i]);
+        add_string(&bloom_filter, argv[i]);
     }
 
     char *last_string = argv[argc - 1];
-    bool match = check_string(bloom_filter, last_string);
+    bool match = check_string(&bloom_filter, last_string);
     printf("Is the last string: '%s' in the set of first strings?\n", last_string);
-    (match == true) ? printf("probably\n") : printf("definitely not\n");
+    printf(match ? "probably\n" : "definitely not\n");
```
Comment (JosephSBoyle) [inline main.c:32]: **praise:** Nice :pray:

**Review by JosephSBoyle — COMMENTED** [review]
Nice changes. I've asked some questions and there's one blocking comment but otherwise it all looks good. :zap:

File: bloomfilter.c (hunk 1 above)
Comment (Lasseone) [reply bloomfilter.c]: > **question (non-blocking):** Should the parameter description mention that the arg is a pointer or is that redundant due to the header containing that information?

I would not mention that it is a pointer, I think that'd be like rewriting part of the code in documentation

File: bitarray.c (hunk 2 above)
Comment (Lasseone) [reply bitarray.c]: I added some explanatory comments, although this is a pretty standard way of reading/writing bits so I wouldn't normally say it should be documented

File: bloomfilter.c (hunk 3 above)
Comment (Lasseone) [reply bloomfilter.c:18]: > **question (non-blocking):** Some questions for my understanding:
> 
>     1. `const char string[]` an array of pointers to characters, and in line 18 we iterate through each character by incrementing the pointer until the value of `character` (the dereferenced pointer) is equal to the terminating '\0' ?
> 
>     2. Is `const char* character = string` initializing `character` as the first element of `string`?
> 
>     3. `character++` Is this how we get the next char pointer?

`const char[]` (when passed as a function argument) is a pointer to the first element in an array of chars. I personally prefer passing - strings as `const char*` intead of `const char[]` just to remove that ambiguity

`const char* character=string` takes a copy of the pointer and stores it as 'character'

`*character` returns what letter is stored at the address that character currently points to

`character++` increments the pointer such that it points to the next character in the string

Comment (JosephSBoyle) [reply bloomfilter.c:18]: Nice, thanks for the explanation!
It's cool that incrementing pointers can be done like that

Comment (Lasseone) [reply bloomfilter.c:18]: Ye under the hood they're just integers which is nice until you realise that x[3] is equivelant to 3[x] lol

**Pull request merged**